_posts/2024-01-25-azure-without-secrets-part-2.md

```diff
@@ -0,0 +1,380 @@
+
+Once again, I put the configuration in two places; `Values` in `local.settings.json` for localhost development, and for the Azure Functions resource, I put it in the app settings using Bicep.
+
+More detailed instructions for the Service Bus queue trigger can be found [here](https://learn.microsoft.com/en-us/azure/azure-functions/functions-identity-based-connections-tutorial-2).
+
+```json
+{
+    "IsEncrypted": false,
+    "Values": {
+        "ServiceBusConnection__fullyQualifiedNamespace": "my-servicebus-namespace.servicebus.windows.net",
+        // ...
+    }
+}
+```
+
+```bicep
+resource functionApp 'Microsoft.Web/sites@2022-09-01' = {
+  name: functionAppName
+  location: location
+  properties: {
+    siteConfig: {
+      appSettings: [
+        {
+          name: 'ServiceBusConnection__fullyQualifiedNamespace'
+          value: '${serviceBus.name}.servicebus.windows.net'
+        }
+      ]
+      // ...
+    }
+    // ...
+  }
+}
+```
+
+### Disable key-based auth (optional)
+You can disable key-based authentication by setting `disableLocalAuth` *true* in the Service Bus namespace resource under `properties` in Bicep.
+
+```bicep
+resource serviceBus 'Microsoft.ServiceBus/namespaces@2021-11-01' = {
+  name: serviceBusNamespaceName
+  location: location
+  properties: {
+    disableLocalAuth: true
+  }
+  // ...
+}
+```
+
+## The list doesn't end here
+In the examples, I showed how to access blobs, SQL database and Service Bus queue with managed identity and RBAC. However, many other services are supported, too. You can find the complete list [here](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/services-id-authentication-support). That list doesn't cover services that are in preview. For example, Azure Cache for Redis [is in preview](https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-azure-active-directory-for-authentication).
+
+In addition to that, even more resources support managed identities. You can, for example, assign Azure API Management managed identity and [access Key Vault using that](https://learn.microsoft.com/en-us/azure/api-management/api-management-howto-use-managed-service-identity).
+
+But wait, there's more! C# isn't the only supported language, the Azure Identity library is available for at least Java, JavaScript, Python and Go as well.
+
+## Closing words
+In this post, I demonstrated how to use system-assigned managed identity to access Azure resources. As you noticed, even though the basic structure of utilizing managed identity and RBAC is the same, each of the services has its little quirks. To get to know these quirks, you need to read the documentation carefully and do some digging. And some scenarios are just not supported (yet). But even if it takes a little more time, I think it's worth it. When you don't have to manage secrets, there's one less place to make possibly costly mistakes.
```

Comment:
Lause
"And some scenarios are just not supported (yet). But even if it takes a little more time, I think it's worth it."
jättää tiukan cliffhangerin lukijalle :) Saiskohan tuohon jonku linki tai selityksen, että mitä ei ole tuettu. Taikka sitte ottais kokonaan pois.

Reply:
Mmmh koitan keksiä siihe jotai tai sit otan pois. Tämänhä on tosiaan tarkotus olla semmonen "getting started" -tyylinen ni perusjutut onnistuu kyllä. Mut niitä ei oo minusta mitenkää yhteen paikkaan kerätty niitä et mitä ei oo tuettu, koska se riippuu niin paljon siitä et mikä resurssi on kyseessä.

Reply:
Otin pois